src/vs/platform/dialogs/electron-main/dialogMainService.ts

```diff
@@ -48,14 +50,13 @@ export class DialogMainService implements IDialogMainService {
 
 	private static readonly workingDirPickerStorageKey = 'pickerWorkingDir';
 
-	private readonly mapWindowToDialogQueue: Map<number, Queue<void>>;
-	private readonly noWindowDialogQueue: Queue<void>;
+	private readonly windowDialogLocks = new Map<number, Set<number>>();
+	private readonly windowDialogQueues = new Map<number, Queue<any>>();
+	private readonly noWindowDialogueQueue = new Queue<any>();
```

Comment:
Can we still get rid of these 2 `any` usages somehow?

Reply:
Not quite sure of the best approach for this. I've attempted to use `MessageBoxReturnValue | SaveDialogReturnValue | OpenDialogReturnValue` for the queues but have had some trouble with getting them to work in `getWindowDialogQueue`.

I don't have a lot of experience with TS generics so I might be missing something really obvious here.

Open to any suggestions.